---
title: "Chang B6 96hr 4wk 2DG Summary Prefrontal Cortex"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
   
---


# Summary

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

```

```{r load packages}
needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gProfileR","rlist", "pheatmap")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","WGCNA_source.R"))
```

There were 16 samples (4 per time and treatment). Mice were fed 2DG (6g/L) or nothing in their water for 96hrs or 2 weeks. WGCNA analysis was performed to determine gene clustering. A soft-thresholding power was chosen so that the scale-free topology correlation attempts to hit 0.9. The power chosen for prefrontal cortex was 15, which allowed the correlations to hover around 0.80. This power clustered the genes into 263 modules originally, however, modules that were highly correlated (0.75 or higher) were further merged together to create 36 modules.

```{r}
modules<-read.csv(here("Data","Prefrontal Cortex","log.tdata.FPKM.sample.info.subset.prefrontal.cortex.WGCNA.module.membership.csv"), header=T)

module_barplot(modules)
```

Pathways were assessed for each module. Out of 36 modules, only 24 modules had pathway hits of any kind. The pathways that had the highest frequency across modules were metabolism of RNA and RNA polymerase II transcription, as well as glucose metabolism. The aquamarine, bisque2, black, darkolivegreen2, darkolivegreen4, hotpink3, and salmon modules all contained these pathways.

## Aquamarine

14 Pathways were identified in the **Aquamarine** module. The pathways were related to mRNA, gene expression and transcription, SOMUylation, and RNA polymerase II.

## Bisque2

15 modules were identified in the **Bisque2** module. The pathways were related to gene transcription, protein processing in the ER, RNA polymerase II, PERK, mRNA, and RNA transport.

## Black

120 genes were identified in the **Black** module. The pathways were related to APC, cell cycle, DNA, metabolism, mitochondrial metabolism, nonsense mediated decay, oxidative phosphorylation, rRNA, RNA polymerase, TCA cycle,  transcriptional regulation, translesion, and ubiquitin.

## Blanchedalmond

9 pathways were identified in the **Blanchedalmond** module. The pathways were related to various aspects of cancer, such as ECM-interaction, calcium signaling, wnt signaling, and tight junctions.

## Blueviolet

6 pathways were identified in the **Blueviolet** module. The pathways were related to all aspects of collagen.

## Chocolate4

169 pathways were identified in the **Chocolate4** module. The pathways were related to AMPK signaling, autophagy, axon guidance, cAMP, cardiac function, chemokine signaling, cardiac function, estrogen, gycosaminoglycan biosynthesis, glucagon signaling, insulin, mTOR signaling, MAPK, MyD88, neuronal, cancer, platelets, signaling, T cell receptors, Th1 and 2 differentiation, Th17 differentiation, toll like receptors, and vasopressin and VEGF. 

## Cornsilk

2 pathways were identified in the **Cornsilk** module. The pathways were Th1 and Th2 cell differentiation and Th17 cell differentiation.

## Cornsilk2

120 pathways were identified in the **Cornsilk2** module. The pathways were related to adaptive immunity, NF-kappa B, APC, antigen processing-cross presentation, cell cycle, cellular responses, interleukin-1, MAPK, p53, signaling, transcriptional regulation, and ubiquitin.

## Darkolivegreen2

35 pathways were identified in the **Darkolivegreen2** module. The pathways were related to neurodegenerative diseases, metabolism, mitochondrial metabolism, oxidative phosphorylation, post-translational protein modification, respiratory electron transport, TCA cycle, thermogenesis, and translation.

## Darkolivegreen4

35 pathways were identified in the **Darkolivegreen4** module. The pathways were related to mRNA, glucose metabolism, SUMOylation, and transport.

## Darkturquoise

27 pathways were identified in the **Darkturquoise** module. The pathways were related to cAMP, GPCR, cortisol, cardiac function, platelets, purine metabolism, MAPK, and signal transduction.

## Dodgerblue4

18 pathways were identified in the **Dodgerblue4** module. The pathways were related to oxidative phosphorylation, neurodegenerative disease, RNA transport, TCA cycle, thermogenesis, and vesicle-mediate transport.

## Grey

5 pathways were identified in the **Grey** module. The pathways were related to gamma-carboxylation and cell cycle.

## Hotpink3

7 pathways were identified in the **Hotpink3** module. The pathways were related to glycolysis, glyconeogenesis, carbon metabolism, amino acids, and glucose metabolism.

## Khaki3

34 pathways were identified in the **Khaki3** module. The pathways were related to collagen, extracellular matrix, MET, protein digestion and absorption, smooth muscle contraction, and TGF-beta signaling.

## Lightblue

2 pathways were identified in the **Lightblue** module. The pathways were Metabolism of angiotensinogen to angiotensins and peptide hormone metabolism.

## Lightgoldenrod3

44 pathways were identified in the **Lightgoldenrod3** module. The pathways were related to MAPK, mTOR, cancer, B cell receptor signaling, protein processing in the ER, VEGF signaling, T cell receptor signaling, and vitamin digestion and absorption.

## Lightpink1

11 pathways were identified in the **Lightpink1** module. The pathways were related to cancer, MAPK signaling, type 2 diabetes, osteoclast differentiation, and wnt signaling.

## Lightskyblue

7 pathways were identified in the **Lightskyblue** module. The pathways were related to estrogen, steroids, and vitamin D metabolism.

## Mediumorchid1

9 pathways were identified in the **Mediumorchid1** module. The pathways were related to digestion, carbohydrate digestion, pancreatice and salivary secretions.

## Pink

37 pathways were identified in the **Pink** module. The pathways were related to innate immunity, natural killer cells, neutrophils, NF-kapp B, NOD-like receptors, toll like receptors, and osteoclast differentiation.

## Salmon

46 pathways were identified in the **Salmon** module. The pathways were related to amino acid metabolism, AMPK, drug metabolism, fatty acid metabolism, gluconeogensis, glycolysis, gluconeogenesis, metabolism, sodium channels, and transport.

```{r}
tdata.FPKM.sample.info <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_phenotypes.RData"))

tdata.FPKM <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_numeric.RData"))

log.tdata.FPKM <- log(tdata.FPKM + 1)
log.tdata.FPKM <- as.data.frame(log.tdata.FPKM)

log.tdata.FPKM.sample.info <- cbind(log.tdata.FPKM, tdata.FPKM.sample.info[,27238:27240])

log.tdata.FPKM.sample.info <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.subset <- log.tdata.FPKM[,colMeans(log.tdata.FPKM != 0) > 0.5] 

log.tdata.FPKM.subset <- log.tdata.FPKM.subset %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.sample.info.subset.prefrontal.cortex <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(Tissue == "Pre-frontal Cortex") %>% column_to_rownames()

log.tdata.FPKM.subset <- subset(log.tdata.FPKM.sample.info.subset.prefrontal.cortex, select = -c(Time,Treatment,Tissue))
  
WGCNA.pathway <-readRDS(here("Data","Prefrontal Cortex","Chang_B6_96hr_4wk_gprofiler_pathway_annotation_list_prefrontal_cortex_WGCNA.RData"))

Matched<-readRDS(here("Data","Prefrontal Cortex","Annotated_genes_in_prefrontal_cortex_WGCNA_Chang_B6_96hr_4wk.RData"))


module.names <- Matched$X..Module.
name <- str_split(module.names,"_")
samples <-c()
for(i in 1:length(name)){
samples[[i]] <- name[[i]][2]
}  
name <- str_split(samples,"\"")
name <- unlist(name)

Treatment <- unclass(as.factor(log.tdata.FPKM.sample.info.subset.prefrontal.cortex[,27238]))
Time <- unclass(as.factor(log.tdata.FPKM.sample.info.subset.prefrontal.cortex[,27237]))
Treat.Time <- paste0(Treatment, Time)
phenotype <- data.frame(cbind(Treatment, Time, Treat.Time))

nSamples <- nrow(log.tdata.FPKM.sample.info.subset.prefrontal.cortex)

MEs0 <- read.csv(here("Data","Prefrontal Cortex","log.tdata.FPKM.sample.info.subset.prefrontal.cortex.WGCNA.module.eigens.csv"),header = T, row.names = 1)
name <- str_split(names(MEs0),"_")
samples <-c()
for(i in 1:length(name)){
samples[[i]] <- name[[i]][2]
}  
name <- str_split(samples,"\"")
name <- unlist(name)
colnames(MEs0) <-name
MEs <- orderMEs(MEs0)
moduleTraitCor <- cor(MEs, phenotype, use = "p");
moduleTraitPvalue <- corPvalueStudent(moduleTraitCor, nSamples)
```

## Relationship between Modules and Traits
```{r, results='asis'}
#sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
                        signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)

# Display the correlation values within a heatmap plot
heat <- pheatmap(moduleTraitCor, main = paste("Module-trait relationships"), display_numbers = textMatrix,color=colorRampPalette(brewer.pal(n = 11, name = "Spectral"))(10), cluster_rows = F, cluster_cols = F, fontsize_number = 4, angle_col = 45, number_color = "black", border_color = "white")
heat
```

## Treatment

The top 10 genes that were significant for treatment within the **limegreen** module and had the highest correlation with limegreen were:

1. BC002059
2. ENSMUSG00000045928
3. Tmem165
4. Klhl26
5. Ado
6. ENSMUSG00000097444
7. Cyp2j6
8. ENSMUSG00000050625
9. ENSMUSG00000084370
10. Slc25a44

The top 10 genes that were significant for treatment within the **rosybrown2** module and had the highest correlation with rosybrown2 were:

1. Uimc1
2. Aars
3. ENSMUSG00000039358
4. Klhdc8b
5. Eps8l2
6. Cym
7. Ttll4
8. Gpr81
9. Alkbh6
10. Cadm3

The top 10 genes that were significant for treatment within the **lavenderblush2** module and had the highest correlation with lavenderblush2 were:

1. Smo
2. Rpn2
3. Rassf2
4. Zfp945
5. Grrp1
6. Rnaset2b
7. Rgl3
8. Bloc1s5
9. Sema4d
10. ENSMUSG00000064982

The top 10 genes that were significant for treatment within the **antiquewhite** module and had the highest correlation with antiquewhite were:

1. Fam19a2
2. ENSMUSG00000028584
3. Tmem86b
4. Ctnnbip1
5. Gpc1
6. Rps6kb2
7. Cxcr3
8. Arhgap25
9. Khdrbs3
10. Ube2q1

The top 10 genes that were significant for treatment within the **pink** module and had the highest correlation with pink were:

1. ENSMUSG00000028876
2. Zzz3
3. Bcl3
4. Pitpnc1
5. Ass1
6. ENSMUSG00000097387
7. 4930426L09Rik
8. Camk2g
9. ENSMUSG00000071516
10. ENSMUSG00000092452

## Time

The top 10 genes that were significant for time within the **lightpink1** module and had the highest correlation with lightpink1 were:

1. Chd3
2. Mpp2
3. Nop10
4. 36598
5. 9330158H04Rik
6. Rdh1
7. Rer1
8. Cdc25c
9. Psmg1
10. Dcst1

The top 10 genes that were significant for time within the **goldenrod3** module and had the highest correlation with goldenrod3 were:

1. Sbk1
2. Gm3194
3. Rad23a
4. Gm5822
5. Prickle1
6. Afap1l2
7. Alox12
8. ENSMUSG00000079644
9. Lurap1
10. Aqp11

The top 10 genes that were significant for time within the **aquamarine** module and had the highest correlation with aquamarine were:

1. ENSMUSG00000052295
2. Prelid1
3. Fbxl2
4. Sytl2
5. Xpa
6. Adpgk
7. C130071C03Rik
8. Repin1
9. Lix1l
10. Psma5

## Treatment by time

The top 10 genes that were significant for treatment by time within the **limegreen** module and had the highest correlation with limegreen were:

1. BC002059
2. Mrpl44
3. ENSMUSG00000045928
4. Tmem165
5. Tmem109
6. Klhl26
7. ENSMUSG00000097444
8. Cyp2j6
9. ENSMUSG00000050625
10. ENSMUSG00000084370

The top 10 genes that were significant for treatment by time within the **mistyrose** module and had the highest correlation with mistyrose were:

1. ENSMUSG00000041165
2. Zfp652
3. Cdk12
4. Lrrfip1
5. Ciapin1
6. Lhfpl2
7. Nisch
8. Alas1
9. Sall2
10. Nhsl2

The top 10 genes that were significant for treatment by time within the **rosybrown2** module and had the highest correlation with rosybrown2 were:

1. Uimc1
2. Gm20469
3. Aars
4. Itgav
5. Klhdc8b
6. Eps8l2
7. Cym
8. Gpr81
9. Alkbh6
10. Cadm3

The top 10 genes that were significant for treatment by time within the **lavenderblush2** module and had the highest correlation with lavenderblush2 were:

1. Rorc
2. Rpn2
3. Zfp945
4. Grrp1
5. Rnaset2b
6. Rgl
7. Bloc1s5
8. Sema4d
9. Col6a5
10. Zmynd10

The top 10 genes that were significant for treatment by time within the **antiquewhite** module and had the highest correlation with antiquewhite were:

1. Rad51ap1
2. Fam19a2
3. ENSMUSG00000028584
4. Tmem86b
5. Ctnnbip1
6. Gpc1
7. Cxcr3
8. Arhgap25
9. Khdrbs3
10. Ube2q1

The top 10 genes that were significant for treatment by time within the **pink** module and had the highest correlation with pink were:

1. Eml6
2. Ewsr1
3. Gabarapl1
4. ENSMUSG00000028876
5. ENSMUSG00000013418
6. ENSMUSG00000061520
7. Zzz3
8. ENSMUSG00000059887
9. Crebzf
10. Pitpnc1
