---
title: "Chang B6 96hr 4wk 2DG Summary Brain"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
   
---


# Summary

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

```

```{r load packages}
needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gProfileR","rlist", "pheatmap")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","WGCNA_source.R"))
```

There were 48 samples (4 per time and treatment for each tissue). Mice were fed 2DG (6g/L) or nothing in their water for 96hrs or 2 weeks. WGCNA analysis was performed to determine gene clustering. A soft-thresholding power was chosen so that the scale-free topology correlation attempts to hit 0.9. The power chosen for hippocampus, hypothalamus, and pre-frontal cortex was 9, which allowed the correlations to hover around 0.88. This power clustered the genes into 195 modules originally, however, modules that were highly correlated (0.75 or higher) were further merged together to create 41 modules.

```{r}
modules<-read.csv(here("Data","Brain","log.tdata.FPKM.sample.info.subset.hip.hyp.cortex.WGCNA.module.membership.csv"), header=T)

module_barplot(modules)
```

Pathways were assessed for each module. Out of 41 modules, only 28 modules had pathway hits of any kind. The pathways that had the highest frequency across modules were metabolism of proteins, RNA, and arachidonic acid. The brown3, cyan, darkgrey, green3, green4, honeydew, lavenderblush1, lightcoral, magenta1, and orangered4 modules all contained these pathways.


## Antiquewhite

60 pathways were identified for the **antiquewhite** module. The pathways identified were involved in amino acid metabolism, cell signaling, neuronal signaling, insulin regulation, cancer, and immunity.

## Blanchedalmond

17 pathways were identified for the **blanchedalmond** module. The pathways identified were involved in neuronal signaling and calcium signaling.

## Brown3

38 pathways were identified for the **brown3** module. The pathways identified were involved in metabolism (macronutrients, TCA cycle, mitochondrial metabolism, respiratory electron transport), thermogenesis, and neurodegenerative diseases.

## Cyan

102 pathways were identified for the **cyan** module. The pathways identified were involved in cell cycle, metabolism, translation, rRNA, ribosomal, and ubiquitin.

## Darkgrey

50 pathways were identified for the **darkgrey** module. The pathways identified were involved in cell cycle, neurodegenerative diseases, thermogenesis, RNA, and cellular responses to stress and external stimuli.

## Darkseagreen

52 pathways were identified for the **darkseagreen** module. The pathways were related to cAMP, insulin signaling, interleukin-17, cell-cell interactions, MAPK, MyD88, p53, T cell receptor signaling, and TLR cascades.

## Darkslateblue

1 pathway (arrythmogenis right ventricular cardiomyopathy) was identified for the **darkslateblue** module.

## Deeppink2

7 pathways were identified for the **deeppink2** module. The pathways identified were involved in signaling and osteoclast differentiation.

## Green3

24 pathways were identified in the **green3** module. The pathways were related to autoimmune diseases, mRNA, mitochondrial metabolism, rRNA, and translation.

## Green4

5 pathways were identified in the **green4** module. The pathways identified were involved in mRNA.

## Grey

17 pathways were identified in the **grey** module. The pathways were related to immunity, complement, bile acids, gamma-carboxylation, steroid hormones, and phenylalanine metabolism.

## Honeydew

22 pathways were identified in the **honeydew** module. The pathways were related to RNA polymerase II, mRNA, SUMOylation, and PERK.

## Lavenderblush1

183 pathways were identified in the **lavenderblush1** module. The pathways related to APC, cellular responses, antigen processing-cross presentation, MHC I mediated antigen processing and presentation, DNA, metabolism, mitochrondrial metabolism, nonsense mediated decay, p53, RNA polymerase I and II, RUNX3, rRNA, transcriptional regulation, thermogenesis, and TCR signaling.

## Lavenderblush2

3 pathways were identified in the **lavenderblush2** module: fatty acids; incretin synthesis, secretion, and activation: and sythesis, secretion and inactivation of glucagon-like peptide-1 (GLP-1)

## Lavenderblush3

1 pathway (axon guidance) was identified in the **lavenderblush3** module.

## Lightblue1

35 pathways were identified in the **lightblue1** module. The pathways were related to beta oxidation, antigen processing and presentation, fatty acid metabolism, MET, extracellular matrix, mitochondrial metabolism, and amino acid metabolism.

## Lightcoral

42 pathways were identified in the **lightcoral** module. The pathways were related to amino acids, AMPK, fatty acid metabolism, gluconeogenesis, glycolysis, insulin resistance, metabolism, sodium/chloride transporters, PPAR signaling, and transport.

## Magenta1

23 pathways were identified in the **magenta1** module. The pathways were related to collagen, extracellular matrix, fatty acid metabolism, protein digestion and absorption, and arachidonic acid metabolism.

## Magenta3

18 pathways were identified in the **magenta3** module. The pathways were related to calcium and calmodulin.

## Magenta4

5 pathways were identified in the **magenta4** module. The pathways identified were related to RNA transcription and Herpes simplex virus 1 infection.

## Mediumorchid3

9 pathways were identified in the **mediumorchid3** module. The pathways identified were related to potassium and calcium channels, and GLP-1 regulates insulin secretion.

## Mediumpurple

43 pathways were identified in the **medium orchid3** module. The pathways were related to axon guidance, growth hormone, MAPK signaling, thyroid hormone signaling, TLR signaling, VEGF, and osteoclast differentiation.

## Orange3

6 pathways were identified in the **orange3** module. The pathways identified were related to muscle contraction.

## Orangered1

2 pathways were identified in the **orangered1** module. The pathways identified were herpes simplex virus 1 infection and ubiquitin mediated proteolysis.

## Orangered4

16 pathways were identified in the **orangered4** module. The pathways were related to keratan, complement, sodium/chloride transporter, platelet degranulation, and arachidonic acid metabolism.

## Plum

3 pathways were identified in the **plum** module. The pathways identified were ether lipid metabolism, metabolic pathways, and sphingolipid metabolism.

## Royalblue

62 pathways were identified in the **royalblue** module. The pathwayd were related to adaptive immunity, B cell receptor signaling, chemokine receptors, complement, cytokine-cytokine, NF-kappa B, NOD-like receptor signaling, TLR cascade, natural killer cells, and autoimmune disease.

## Thistle3

14 pathways were identified in the **thistle3** module. The pathways were related to defensins, digestion, immune system, neutrophil degranulation, NOD-like receptor signaling, and pancreatice secretion.

```{r}
tdata.FPKM.sample.info <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_phenotypes.RData"))

tdata.FPKM <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_numeric.RData"))

log.tdata.FPKM <- log(tdata.FPKM + 1)
log.tdata.FPKM <- as.data.frame(log.tdata.FPKM)

log.tdata.FPKM.sample.info <- cbind(log.tdata.FPKM, tdata.FPKM.sample.info[,27238:27240])

log.tdata.FPKM.sample.info <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.subset <- log.tdata.FPKM[,colMeans(log.tdata.FPKM != 0) > 0.5] 

log.tdata.FPKM.subset <- log.tdata.FPKM.subset %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.sample.info.subset.hip.hypo.cortex <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(Tissue %in% c("Hippocampus","Hypothanamus","Pre-frontal Cortex")) %>% column_to_rownames()

log.tdata.FPKM.subset <- subset(log.tdata.FPKM.sample.info.subset.hip.hypo.cortex, select = -c(Time,Treatment,Tissue))
  
WGCNA.pathway <-readRDS(here("Data","Brain","Chang_B6_96hr_4wk_gprofiler_pathway_annotation_list_hip_hyp_cortex_WGCNA.RData"))

Matched<-readRDS(here("Data","Brain","Annotated_genes_in_hip_hyp_cortex_WGCNA_Chang_B6_96hr_4wk.RData"))


module.names <- Matched$X..Module.
name <- strsplit(module.names,"_")
samples <-c()
for(i in 1:length(name)){
samples[[i]] <- name[[i]][2]
}  
name <- strsplit(samples,"\"")
name <- unlist(name)

Treatment <- unclass(as.factor(log.tdata.FPKM.sample.info.subset.hip.hypo.cortex[,27238]))
Tissue <- unclass(as.factor(log.tdata.FPKM.sample.info.subset.hip.hypo.cortex[,27239]))
Time <- unclass(as.factor(log.tdata.FPKM.sample.info.subset.hip.hypo.cortex[,27237]))
Treat.Tissue <- paste0(Treatment,Tissue)
Treat.Time <- paste0(Treatment, Time)
Tissue.Time <- paste0(Tissue, Time)
Treat.Tissue.Time <- paste0(Treatment,Tissue, Time)
phenotype <- data.frame(cbind(Tissue, Treatment, Time, Treat.Tissue, Treat.Time, Tissue.Time, Treat.Tissue.Time))

nSamples <- nrow(log.tdata.FPKM.sample.info.subset.hip.hypo.cortex)

MEs0 <- read.csv(here("Data","Brain","log.tdata.FPKM.sample.info.subset.hip.hyp.cortex.WGCNA.module.eigens.csv"),header = T, row.names = 1)
name <- strsplit(names(MEs0),"_")
samples <-c()
for(i in 1:length(name)){
samples[[i]] <- name[[i]][2]
}  
name <- strsplit(samples,"\"")
name <- unlist(name)
colnames(MEs0) <-name
MEs <- orderMEs(MEs0)
moduleTraitCor <- cor(MEs, phenotype, use = "p");
moduleTraitPvalue <- corPvalueStudent(moduleTraitCor, nSamples)
```

## Relationship between Modules and Traits
```{r, results='asis'}
#sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
                        signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)

# Display the correlation values within a heatmap plot
heat <- pheatmap(moduleTraitCor, main = paste("Module-trait relationships"), display_numbers = textMatrix,color=colorRampPalette(brewer.pal(n = 11, name = "Spectral"))(10), cluster_rows = F, cluster_cols = F, fontsize_number = 4, angle_col = 45, number_color = "black", border_color = "white")
heat
```

## Tissue

The top 10 genes that were significant for tissue within the **lavenderblush3** module and had the highest correlation with lavenderblush3 were:

1. lno80e
2. Rab7
3. ld4
4. 4632415L05Rik
5. Gtf3c4
6. Tagln2
7. Map3k19
8. Mlc1
9. Gm6981
10. ENSMUSG00000086732

The top 10 genes that were significant for tissue within the **darkslateblue** module and had the highest correlation with darkslateblue were:

1. Micall2
2. Celf2
3. Pygl
4. Rufy4
5. Nos1ap
6. Brinp1
7. Nfe2l3
8. ENSMUSG00000026805
9. Gm3604
10. Gm8237

When assessing the overall sample contribution through their summary eigengenes it appears that for **darkslateblue** all hippocampus had a positive effect while hypothalamus and prefrontal cortex had a negative effect. Hypothalumus and prefrontal cortex had under-expressed genes while hippocampus had over-expressed genes.

The top 10 genes that were significant for tissue within the **chocolate3** module and had the highest correlation with chocolate3 were:

1. Rpl10-ps3
2. Bcl7a
3. Psma2
4. Rpl14
5. Med1
6. Laptm5
7. Hsdl2
8. Rit2
9. Klhdc9
10. Mark3

## Treatment

The top 10 genes that were significant for treatment within the **royalblue** module and had the highest correlation with royalblue were:

1. Mb21d2
2. Vmn1r-ps128
3. Pola1
4. Elac1
5. Fras1
6. ENSMUSG00000037469
7. Slc39a11
8. Ctps
9. Pkd2
10. ENSMUSG00000083819

When assessing the overall sample contribution through their summary eigengenes it appears that for **royalblue** most controls had a negative effect while most 2DG had a positive effect. Controls had under-expressed genes while 2DG had over-expressed genes.

## Time

None

## Treatment by tissue

The top 10 genes that were significant for treatment by tissue within the **royalblue** module and had the highest correlation with royalblue were:

1. Gdi1
2. Gucy1b3
3. Vmn1r-ps128
4. Btbd9
5. Pola1
6. Trim35
7. Elac1
8. Pelo
9. Ly6e
10. Pkd2

When assessing the overall sample contribution through their summary eigengenes it appears that for **royalblue** most controls had a negative effect while most 2DG had a positive effect, independent of tissue. Controls had under-expressed genes while 2DG had over-expressed genes. 

## Treatment by time

The top 10 genes that were significant for treatment by time within the **royalblue** module and had the highest correlation with royalblue were:

1. Snx19
2. Mb21d2
3. Gdi1
4. Vmn1r-ps128
5. Clptm1
6. Pola1
7. Trim35
8. Elac1
9. Fras1
10. Pelo

When assessing the overall sample contribution through their summary eigengenes it appears that for **royalblue** most controls had a negative effect while most 2DG had a positive effect, independent of time. Controls had under-expressed genes while 2DG had over-expressed genes.

## Tissue by time

The top 10 genes that were significant for tissue by time within the **lavenderblush3** module and had the highest correlation with lavenderblush3 were:

1. Herc2
2. Ino80e
3. Rab7
4. Id4
5. 4632415L05Rik
6. Gtf3c4
7. Tagln4
8. Map3k19
9. Mlc1
10. Amer2

The top 10 genes that were significant for tissue by time within the **darkslateblue** module and had the highest correlation with darkslateblue were:

1. Micall2
2. Celf2
3. Anks1b
4. ENSMUSG00000097393
5. Pygl
6. Rufy4
7. Nos1ap
8. Brinp1
9. Nfe2l3
10. ENSMUSG00000026805

When assessing the overall sample contribution through their summary eigengenes it appears that for **darkslateblue** all hippocampus had a positive effect while hypothalamus and prefrontal cortex had a negative effect, all independent of time. Hypothalumus and prefrontal cortex had under-expressed genes while hippocampus had over-expressed genes.

The top 10 genes that were significant for tissue by time within the **chocolate3** module and had the highest correlation with chocolate3 were:

1. Bcl7a
2. Psma2
3. Rpl14
4. Med1
5. Laptm5
6. Hsdl2
7. Rit2
8. Klhdc9
9. Mark3
10. Hcrtr2

## Treatment by tissue by time

The top 10 genes that were significant for treatment by tissue by time within the **royalblue** module and had the highest correlation with royalblue were:

1. Gdi1
2. Gucy1b3
3. Vmn1r-ps128
4. Btbd9
5. Pola1
6. Trim35
7. Elac1
8. Pelo
9. Ly6e
10. Pkd2

When assessing the overall sample contribution through their summary eigengenes it appears that for **royalblue** most controls had a negative effect while most 2DG had a positive effect, independent of tissue and time. Controls had under-expressed genes while 2DG had over-expressed genes.