---
title: "Hub Genes"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
---

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gProfileR", "WGCNA", "xtable", "pander", "kableExtra","downloadthis")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","plot_theme.R"))
source(here("source_files","WGCNA_source.R"))
```

```{r load data}
tdata.FPKM.sample.info <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_phenotypes.RData"))

tdata.FPKM <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_numeric.RData"))

log.tdata.FPKM <- log(tdata.FPKM + 1)
log.tdata.FPKM <- as.data.frame(log.tdata.FPKM)

log.tdata.FPKM.sample.info <- cbind(log.tdata.FPKM, tdata.FPKM.sample.info[,27238:27240])

log.tdata.FPKM.sample.info <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.subset <- log.tdata.FPKM[,colMeans(log.tdata.FPKM != 0) > 0.5] 

log.tdata.FPKM.sample.info.subset <- cbind(log.tdata.FPKM.subset,tdata.FPKM.sample.info[,27238:27240])
log.tdata.FPKM.sample.info.subset <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.sample.info.subset <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% column_to_rownames()

module.labels <- readRDS(here("Data","log.tdata.FPKM.sample.info.subset.WGCNA.module.labels.RData"))

modules <- readRDS(here("Data","log.tdata.FPKM.sample.info.subset.WGCNA.module.membership.RData"))

net.deg <- readRDS(here("Data","Chang_2DG_BL6_connectivity.RData"))

ensembl.location <- readRDS(here("Data","Ensembl_gene_id_and_location.RData"))

Matched<-readRDS(here("Data","Annotated_genes_in_WGCNA_Chang_B6_96hr_4wk.RData"))
```

## Hub Genes {.tabset .tabset-fade .tabset-pills}

Hub genes are genes in the module network that have the highest connectivity. For instance, if the gene significance of a particular module is associated with treatment the hub genes in this module are likely associated with this trait.

```{r, results='asis'}
hub.genes.list <- hub.genes(net.deg,module.labels,modules,ensembl.location)
```

### Lightyellow Module
```{r}
DT.table.hub(hub.genes.list[[1]])
```

### Turquoise Module
```{r}
DT.table.hub(hub.genes.list[[2]])
```

### Darkgreen Module
```{r}
DT.table.hub(hub.genes.list[[3]])
```

### Darkolivegreen Module
```{r}
DT.table.hub(hub.genes.list[[4]])
```

### Black Module
```{r}
DT.table.hub(hub.genes.list[[5]])
```

### Midnightblue Module
```{r}
DT.table.hub(hub.genes.list[[6]])
```

### Pink Module
```{r}
DT.table.hub(hub.genes.list[[7]])
```

### Skyblue Module
```{r}
DT.table.hub(hub.genes.list[[8]])
```

### Lightcyan Module
```{r}
DT.table.hub(hub.genes.list[[9]])
```

### Darkorange2 Module
```{r}
DT.table.hub(hub.genes.list[[10]])
```

### Mediumpurple3 Module
```{r}
DT.table.hub(hub.genes.list[[11]])
```

### Royalblue Module
```{r}
DT.table.hub(hub.genes.list[[12]])
```

### Lightsteelblue1 Module
```{r}
DT.table.hub(hub.genes.list[[13]])
```

### Violet Module
```{r}
DT.table.hub(hub.genes.list[[14]])
```

### Skyblue3 Module
```{r}
DT.table.hub(hub.genes.list[[15]])
```

### Orangered4 Module
```{r}
DT.table.hub(hub.genes.list[[16]])
```

### Lightcyan1 Module
```{r}
DT.table.hub(hub.genes.list[[17]])
```

### Ivory Module
```{r}
DT.table.hub(hub.genes.list[[18]])
```

### Floralwhite Module
```{r}
DT.table.hub(hub.genes.list[[19]])
```

### Grey Module
```{r}
DT.table.hub(hub.genes.list[[20]])
```

## ANOVA {.tabset .tabset-fade .tabset-pills}

To perform analysis of variance Aligned Rank Transformation was performed, which is designed to handle non-parametric multi-factorial experiments. P-value was adjusted using Benjamini-hochberg procedure. Only the top 10 hub genes are listed.

```{r, results='asis'}
log.tdata.FPKM.sample.info.subset$Time <- as.factor(log.tdata.FPKM.sample.info.subset$Time)
log.tdata.FPKM.sample.info.subset$Treatment <- as.factor(log.tdata.FPKM.sample.info.subset$Treatment)
log.tdata.FPKM.sample.info.subset$Tissue <- as.factor(log.tdata.FPKM.sample.info.subset$Tissue)

for(m in module.labels) {
    name <- str_split(m,"_")[[1]][2]
    hub.met <- net.deg[modules$Gene,] %>%
      dplyr::mutate(Gene=modules$Gene) %>%
      dplyr::filter(modules$Module == m) %>%
      dplyr::arrange(desc(kWithin)) %>%
      dplyr::select(Gene, kWithin)
    gene <- ensembl.location$external_gene_name[match(hub.met$Gene,ensembl.location$ensembl_gene_id)]
    hub.met <- cbind(gene,hub.met)
    cat("\n###",name, "{.tabset .tabset-fade .tabset-pills}","\n")
          for(i in 1:10){
             cat("\n####",hub.met$gene[i],"\n")
             m <- art(data = log.tdata.FPKM.sample.info.subset, log.tdata.FPKM.sample.info.subset[,hub.met$Gene[i]] ~ Time*Treatment*Tissue)
             model <- anova(m)
             adjust <- p.adjust(model$`Pr(>F)`, method = "BH")
             print(knitr::kable(cbind(model,adjust)) %>% 
                      column_spec(8, color = "white", background = spec_color(adjust, end = 0.05, scale_from = c(0,0.05))) %>% kable_styling(bootstrap_options = c("striped", "hover", "responsive")))
             cat("\n \n")
             }
          cat("\n \n")
          }
```