---
title: "Hub Genes Small Intestine"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
---

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gProfileR", "WGCNA", "xtable", "pander", "kableExtra","downloadthis")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","plot_theme.R"))
source(here("source_files","WGCNA_source.R"))
```
```{r load data}
tdata.FPKM.sample.info <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_phenotypes.RData"))

tdata.FPKM <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_numeric.RData"))

log.tdata.FPKM <- log(tdata.FPKM + 1)
log.tdata.FPKM <- as.data.frame(log.tdata.FPKM)

log.tdata.FPKM.sample.info <- cbind(log.tdata.FPKM, tdata.FPKM.sample.info[,27238:27240])

log.tdata.FPKM.sample.info <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.subset <- log.tdata.FPKM[,colMeans(log.tdata.FPKM != 0) > 0.5] 

log.tdata.FPKM.sample.info.subset <- cbind(log.tdata.FPKM.subset,tdata.FPKM.sample.info[,27238:27240])
log.tdata.FPKM.sample.info.subset <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.sample.info.subset.small.intestine <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(Tissue == "Small Intestine") %>% column_to_rownames()

module.labels <- readRDS(here("Data","Small Intestine","log.tdata.FPKM.sample.info.subset.small.intestine.WGCNA.module.labels.RData"))

modules <- readRDS(here("Data","Small Intestine","log.tdata.FPKM.sample.info.subset.small.intestine.WGCNA.module.membership.RData"))

net.deg <- readRDS(here("Data","Small Intestine","Chang_2DG_BL6_connectivity_small_intestine.RData"))

ensembl.location <- readRDS(here("Data","Ensembl_gene_id_and_location.RData"))
```

## Hub Genes {.tabset .tabset-fade .tabset-pills}

Hub genes are genes in the module network that have the highest connectivity. For instance, if the gene significance of a particular module is associated with treatment the hub genes in this module are likely associated with this trait.

```{r, results='asis'}
hub.genes.list <- hub.genes(net.deg,module.labels,modules,ensembl.location)
```

### Cornsilk Module
```{r}
DT.table.hub(hub.genes.list[[1]])
```

### Blue1 Module
```{r}
DT.table.hub(hub.genes.list[[2]])
```

### Rosybrown3 Module
```{r}
DT.table.hub(hub.genes.list[[3]])
```

### Antiquewhite2 Module
```{r}
DT.table.hub(hub.genes.list[[4]])
```

### Aliceblue Module
```{r}
DT.table.hub(hub.genes.list[[5]])
```

### Blue3 Module
```{r}
DT.table.hub(hub.genes.list[[6]])
```

### Coral1 Module
```{r}
DT.table.hub(hub.genes.list[[7]])
```

### Magenta3 Module
```{r}
DT.table.hub(hub.genes.list[[8]])
```

### Burlywood Module
```{r}
DT.table.hub(hub.genes.list[[9]])
```

### Lightblue2 Module
```{r}
DT.table.hub(hub.genes.list[[10]])
```

### Green4 Module
```{r}
DT.table.hub(hub.genes.list[[11]])
```

### Antiquewhite1 Module
```{r}
DT.table.hub(hub.genes.list[[12]])
```

### Sienna3 Module
```{r}
DT.table.hub(hub.genes.list[[13]])
```

### Coral4 Module
```{r}
DT.table.hub(hub.genes.list[[14]])
```

### Blue2 Module
```{r}
DT.table.hub(hub.genes.list[[15]])
```

### Antiquewhite4 Module
```{r}
DT.table.hub(hub.genes.list[[16]])
```

### Firebrick Module
```{r}
DT.table.hub(hub.genes.list[[17]])
```

### Deepskyblue Module
```{r}
DT.table.hub(hub.genes.list[[18]])
```

### Hotpink4 Module
```{r}
DT.table.hub(hub.genes.list[[19]])
```

### Grey Module
```{r}
DT.table.hub(hub.genes.list[[20]])
```

### Lightgreen Module
```{r}
DT.table.hub(hub.genes.list[[21]])
```

### Rosybrown4 Module
```{r}
DT.table.hub(hub.genes.list[[22]])
```

### Royalblue Module
```{r}
DT.table.hub(hub.genes.list[[23]])
```

### Orange Module
```{r}
DT.table.hub(hub.genes.list[[24]])
```

### White Module
```{r}
DT.table.hub(hub.genes.list[[25]])
```

### Slateblue2 Module
```{r}
DT.table.hub(hub.genes.list[[26]])
```

### Magenta2 Module
```{r}
DT.table.hub(hub.genes.list[[27]])
```

### Mediumpurple3 Module
```{r}
DT.table.hub(hub.genes.list[[28]])
```

### Thistle2 Module
```{r}
DT.table.hub(hub.genes.list[[29]])
```

### Yellow4 Module
```{r}
DT.table.hub(hub.genes.list[[30]])
```

### Indianred Module
```{r}
DT.table.hub(hub.genes.list[[31]])
```

### Antiquewhite Module
```{r}
DT.table.hub(hub.genes.list[[32]])
```

### Mediumpurple1 Module
```{r}
DT.table.hub(hub.genes.list[[33]])
```

### Lightblue4 Module
```{r}
DT.table.hub(hub.genes.list[[34]])
```

### Firebrick3 Module
```{r}
DT.table.hub(hub.genes.list[[35]])
```

### Mistyrose Module
```{r}
DT.table.hub(hub.genes.list[[36]])
```

### Moccasin Module
```{r}
DT.table.hub(hub.genes.list[[37]])
```

### Chocolate3 Module
```{r}
DT.table.hub(hub.genes.list[[38]])
```

### Pink2 Module
```{r}
DT.table.hub(hub.genes.list[[39]])
```

### Sienna1 Module
```{r}
DT.table.hub(hub.genes.list[[40]])
```

### Darkgoldenrod4 Module
```{r}
DT.table.hub(hub.genes.list[[41]])
```

### Lightskyblue2 Module
```{r}
DT.table.hub(hub.genes.list[[42]])
```

### Linen Module
```{r}
DT.table.hub(hub.genes.list[[43]])
```

### Oldlace Module
```{r}
DT.table.hub(hub.genes.list[[44]])
```

### Lightblue Module
```{r}
DT.table.hub(hub.genes.list[[45]])
```

### Red4 Module
```{r}
DT.table.hub(hub.genes.list[[46]])
```

## ANOVA {.tabset .tabset-fade .tabset-pills}

To perform analysis of variance Aligned Rank Transformation was performed, which is designed to handle non-parametric multi-factorial experiments. P-value was adjusted using Benjamini-hochberg procedure. Only the top 10 hub genes are listed.

```{r, results='asis'}
log.tdata.FPKM.sample.info.subset.small.intestine$Time <- as.factor(log.tdata.FPKM.sample.info.subset.small.intestine$Time)
log.tdata.FPKM.sample.info.subset.small.intestine$Treatment <- as.factor(log.tdata.FPKM.sample.info.subset.small.intestine$Treatment)

for(m in module.labels) {
    name <- str_split(m,"_")[[1]][2]
    hub.met <- net.deg[modules$Gene,] %>%
      dplyr::mutate(Gene=modules$Gene) %>%
      dplyr::filter(modules$Module == m) %>%
      dplyr::arrange(desc(kWithin)) %>%
      dplyr::select(Gene, kWithin)
    gene <- ensembl.location$external_gene_name[match(hub.met$Gene,ensembl.location$ensembl_gene_id)]
    hub.met <- cbind(gene,hub.met)
    cat("\n###",name, "{.tabset .tabset-fade .tabset-pills}","\n")
          for(i in 1:10){
             cat("\n####",hub.met$gene[i],"\n")
             m <- art(data = log.tdata.FPKM.sample.info.subset.small.intestine, log.tdata.FPKM.sample.info.subset.small.intestine[,hub.met$Gene[i]] ~ Time*Treatment)
             model <- anova(m)
             adjust <- p.adjust(model$`Pr(>F)`, method = "BH")
             print(knitr::kable(cbind(model,adjust)) %>% 
                      column_spec(8, color = "white", background = spec_color(adjust, end = 0.05, scale_from = c(0,0.05))) %>% kable_styling(bootstrap_options = c("striped", "hover", "responsive")))
             cat("\n \n")
             }
          cat("\n \n")
          }
```