---
title: "Differential Expression Hippocampus"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
   
---

# Introduction

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

```

This dataset contains nine tissues (heart, hippocampus, hypothalamus, kidney, liver, prefrontal cortex, skeletal hippocampus, small intestine, and spleen) from C57BL/6J mice that were fed 2-deoxyglucose (6g/L) through their drinking water for 96hrs or 4wks. 96hr mice were given their 2DG treatment 2 weeks after the other cohort started the 4 week treatment. The organs from the mice were harvested and processed for metabolomics and transcriptomics. The data in this document pertains to the transcriptomics data only. The counts that were used were FPKM normalized before being log transformed. It was determined that sample A113 had low RNAseq quality and through further analyses with PCA, MA plots, and clustering was an outlier and will be removed for the rest of the analyses performed. This document will assess the effects of time, treatment, and time-by-treatment interaction for all genes identified within this tissue using ANOVA.

```{r load packages}
needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gprofiler2","rlist", "plotly","kableExtra","foreach","doParallel","doFuture","pheatmap")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","WGCNA_source.R"))
```

```{r load data}
tdata.FPKM.sample.info <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_phenotypes.RData"))

tdata.FPKM <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_numeric.RData"))

log.tdata.FPKM <- log(tdata.FPKM + 1)
log.tdata.FPKM <- as.data.frame(log.tdata.FPKM)

log.tdata.FPKM.sample.info <- cbind(log.tdata.FPKM, tdata.FPKM.sample.info[,27238:27240])

log.tdata.FPKM.sample.info <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.subset <- log.tdata.FPKM[,colMeans(log.tdata.FPKM != 0) > 0.5] 

log.tdata.FPKM.subset <- log.tdata.FPKM.subset %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.sample.info.subset.hippocampus <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(Tissue == "Hippocampus") %>% column_to_rownames()

log.tdata.FPKM.subset <- subset(log.tdata.FPKM.sample.info.subset.hippocampus, select = -c(Time,Treatment,Tissue))

log.tdata.FPKM.subset <- log.tdata.FPKM.subset[,colMeans(log.tdata.FPKM.subset != 0) > 0.5] 

```

```{r ANOVA, cache=TRUE}
log.tdata.FPKM.sample.info.subset.hippocampus$Time <- as.factor(log.tdata.FPKM.sample.info.subset.hippocampus$Time)

log.tdata.FPKM.sample.info.subset.hippocampus$Treatment <- as.factor(log.tdata.FPKM.sample.info.subset.hippocampus$Treatment)

ensembl.location <- readRDS(here("Data","Ensembl_gene_id_and_location.RData"))

frame <- matrix(data = NA, nrow = length(log.tdata.FPKM.subset), ncol = 8)

for(i in 1:length(log.tdata.FPKM.subset)){
  m <- art(data = log.tdata.FPKM.sample.info.subset.hippocampus, log.tdata.FPKM.subset[,i] ~ Time*Treatment)
  
  model <- anova(m)
  adjust <- p.adjust(model$`Pr(>F)`, method = "BH")
  
  gene <- ensembl.location$external_gene_name[match(names(log.tdata.FPKM.subset)[i],ensembl.location$ensembl_gene_id)]

  frame[i,] <- c(gene, names(log.tdata.FPKM.subset)[i], model[,7][2], adjust[2], model[,7][1], adjust[1], model[,7][3],adjust[3])
}

colnames(frame) <- c("External Gene Name", "Gene ID", "Treatment", "BH Treatment","Time", "BH Time", " Treatment by time", "BH Treatment by time")

frame <- as.data.frame(frame)

#Treatment.sig.only <- 
```
```{r}
Treatment.sig.only <- frame %>%
  filter(frame$`BH Treatment` < 0.05) %>%
  filter(`BH Time` > 0.05) %>%
  filter(`BH Treatment by time` > 0.05)

saveRDS(Treatment.sig.only, here("Data","Hippocampus","Chang_2DG_BL6_DE_genes_hippocampus.RData"))
```

## ANOVA
```{r, results='asis'}
DT::datatable(frame, extensions = 'Buttons',
                                           rownames = FALSE, 
                                           filter="top",
                                           options = list(dom = 'Blfrtip',
                                                          buttons = c('copy', 'csv', 'excel'),
                                                          lengthMenu = list(c(10,25,50,-1),
                                                                            c(10,25,50,"All")), 
                                                          scrollX= TRUE), class = "display",
              colnames = c("External Gene Name", "Gene ID", "Treatment", "BH Treatment","Time", "BH Time", " Treatment by time", "BH Treatment by time"))
```
