---
title: "Upset Plot of Differential Expression Analysis"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
   
---

<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/1.10.5/css/jquery.dataTables.min.css">
<script src="https://code.jquery.com/jquery-2.1.2.min.js"></script>
<script src="https://cdn.datatables.net/1.10.5/js/jquery.dataTables.min.js"></script>

<script type="text/javascript">
         $(document).ready(function() {
             $("table").DataTable();
         } );
</script>

# Introduction

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

```

This dataset contains nine tissues (heart, hippocampus, hypothalamus, kidney, liver, prefrontal cortex, skeletal spleen, small intestine, and spleen) from C57BL6/J mice that were fed 2-deoxyglucose (6g/L) through their drinking water for 96hrs or 4wks. 96hr mice were given their 2DG treatment 2 weeks after the other cohort started the 4 week treatment. The organs from the mice were harvested and processed for proteomics, metabolomics, and transcriptomics. The data in this document pertains to the transcriptomics data only. The counts that were used were FPKM normalized before being log transformed. It was determined that sample A113 had low RNAseq quality and through further analyses with PCA, MA plots, and clustering was an outlier and will be removed for the rest of the analyses performed.

```{r load packages}
needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gprofiler2","rlist", "plotly","kableExtra","foreach","doParallel","doFuture","pheatmap","UpSetR")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","WGCNA_source.R"))
```

```{r}
heart <- readRDS(here("Data","Heart","Chang_2DG_BL6_DE_genes_heart.RData"))

hip <- readRDS(here("Data","Hippocampus","Chang_2DG_BL6_DE_genes_hippocampus.RData"))

hyp <- readRDS(here("Data","Hypothalamus","Chang_2DG_BL6_DE_genes_hypothalamus.RData"))

kidney <- readRDS(here("Data","Kidney","Chang_2DG_BL6_DE_genes_kidney.RData"))

liver <- readRDS(here("Data","Liver","Chang_2DG_BL6_DE_genes_liver.RData"))

muscle <- readRDS(here("Data","Skeletal Muscle","Chang_2DG_BL6_DE_genes_muscle.RData"))

intestine <- readRDS(here("Data","Small Intestine","Chang_2DG_BL6_DE_genes_small_intestine.RData"))

cortex <- readRDS(here("Data","Prefrontal Cortex","Chang_2DG_BL6_DE_genes_prefrontal_cortex.RData"))

spleen <- readRDS(here("Data","Spleen","Chang_2DG_BL6_DE_genes_spleen.RData"))
```

```{r}
profiler.heart <- gost(heart$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.hip <- gost(hip$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.hyp <- gost(hyp$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.kidney <- gost(kidney$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.liver <- gost(liver$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.cortex <- gost(cortex$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.muscle <- gost(muscle$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.intestine <- gost(intestine$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)

profiler.spleen <- gost(spleen$`Gene ID`,organism = "mmusculus", correction_method = "fdr", sources = c("KEGG","REAC"), evcodes = T)
```

## Upset Plot {.tabset .tabset-fade .tabset-pills}

Differentially expressed genes were compared across tissues to determine how many genes were shared across tissues. 

### Genes
```{r}
treat.genes <- list("Heart" = heart$`Gene ID`, "Hippocampus" = hip$`Gene ID`, "Hypothalamus" = hyp$`Gene ID`, "Kidney" = kidney$`Gene ID`, "Liver" = liver$`Gene ID`, "Prefrontal Cortex" = cortex$`Gene ID`, "Skeletal Muscle" = muscle$`Gene ID`, "Small Intestine" = intestine$`Gene ID`, "Spleen" = spleen$`Gene ID`)

upset(nsets=9, fromList(treat.genes), main.bar.color = "mediumseagreen", matrix.color = "steelblue4", sets.bar.color = "lightsalmon", text.scale = 1)
```

### Pathways

Overrrepresented pathways were compared across tissues to determine how many pathways were shared across tissues.

```{r}
treat.pathways <- list("Prefrontal Cortex" = profiler.cortex$result$term_name, "Skeletal Muscle" = profiler.muscle$result$term_name, "Small Intestine" = profiler.intestine$result$term_name)

upset(nsets=9, fromList(treat.pathways), main.bar.color = "mediumseagreen", matrix.color = "steelblue4", sets.bar.color = "lightsalmon", text.scale = 1)
```