---
title: "Chang B6 96hr 4wk 2DG Summary Skeletal Muscle, Small Intestine, and Spleen 4wks"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
   
---


# Summary

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

```

```{r load packages}
needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gProfileR","rlist", "pheatmap")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","WGCNA_source.R"))
```

There were 47 samples (4 per time and treatment minus skeletal muscle 96 hr control). Mice were fed 2DG (6g/L) or nothing in their water for 96hrs or 4 weeks. WGCNA analysis was performed to determine gene clustering. A soft-thresholding power was chosen so that the scale-free topology correlation attempts to hit 0.9. The power chosen for skeletal muscle, small intestine, and spleen for 4 wks was 1, which allowed the correlations to hover around 0.98. This power clustered the genes into 5 modules originally, however, modules that were highly correlated (0.75 or higher) were further merged together to create 4 modules.

```{r}
modules<-read.csv(here("Data","SM.SI.spleen.4wks","log.tdata.FPKM.sample.info.subset.SM.SI.spleen.4wks.WGCNA.module.membership.csv"), header=T)

module_barplot(modules)
```

Pathways were assessed for each module. Out of 4 modules, only 3 modules had pathway hits of any kind. There were many pathways that had the highest frequency across modules, including class 1 MHC, fructose and mannose metabolism, and many metabolism pathways. The blue, brown, and turquoise modules all contained these pathways.

## Blue

116 pathways were identified for the **blue** module. The pathways identified were involved in fatty acid metabolism, fructose and mannose metabolism, glucagon signaling, mitochondrial pathways, pentose phosphate pathways, and purine pathways.

## Brown

81 pathways were identified for the **brown** module. The pathways identified were involved in fat digestion and absorption, fructose pathways, glutathione metabolism, various lipid metabolisms, amino acid metabolisms, vitamin and mineral metabolism, hormone metabolism, and neuronal pathways. 

## Turquoise

315 pathways were identified for the **turquoise** module. The pathways identified were involved in Alzheimer's disease, several immune pathways, cell cycle, external cellular response (stress, heat stress), chromatin, TCA cycle, and multiple DNA and RNA pathways. 

```{r}
tdata.FPKM.sample.info <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_phenotypes.RData"))

tdata.FPKM <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_numeric.RData"))

log.tdata.FPKM <- log(tdata.FPKM + 1)
log.tdata.FPKM <- as.data.frame(log.tdata.FPKM)

log.tdata.FPKM.sample.info <- cbind(log.tdata.FPKM, tdata.FPKM.sample.info[,27238:27240])

log.tdata.FPKM.sample.info <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.subset <- log.tdata.FPKM[,colMeans(log.tdata.FPKM != 0) > 0.5] 

log.tdata.FPKM.subset <- log.tdata.FPKM.subset %>% rownames_to_column() %>% filter(rowname != "A113") %>% column_to_rownames()

log.tdata.FPKM.sample.info.subset.SM.SI.spleen.4wks <- log.tdata.FPKM.sample.info %>% rownames_to_column() %>% filter(Tissue %in% c("Skeletal Muscle", "Spleen", "Small Intestine")) %>% filter(Time =="4 wks") %>% column_to_rownames()

log.tdata.FPKM.subset <- subset(log.tdata.FPKM.sample.info.subset.SM.SI.spleen.4wks, select = -c(Time,Treatment,Tissue))
  
WGCNA.pathway <-readRDS(here("Data","SM.SI.spleen.4wks","Chang_B6_96hr_4wk_gprofiler_pathway_annotation_list_SM_SI_spleen_4wks_WGCNA.RData"))

Matched<-readRDS(here("Data","SM.SI.spleen.4wks","Annotated_genes_in_SM_SI_spleen_4wks_WGCNA_Chang_B6_96hr_4wk.RData"))


module.names <- Matched$X..Module.
name <- strsplit(module.names,"_")
samples <-c()
for(i in 1:length(name)){
samples[[i]] <- name[[i]][2]
}  
name <- strsplit(samples,"\"")
name <- unlist(name)

Treatment <- unclass(as.factor(log.tdata.FPKM.sample.info.subset.SM.SI.spleen.4wks[,27238]))
Tissue <- unclass(as.factor(log.tdata.FPKM.sample.info.subset.SM.SI.spleen.4wks[,27239]))
Treat.Tissue <- paste0(Treatment,Tissue)

phenotype <- data.frame(cbind(Tissue, Treatment,Treat.Tissue))

nSamples <- nrow(log.tdata.FPKM.sample.info.subset.SM.SI.spleen.4wks)

MEs0 <- read.csv(here("Data","SM.SI.spleen.4wks","log.tdata.FPKM.sample.info.subset.SM.SI.spleen.4wks.WGCNA.module.eigens.csv"),header = T, row.names = 1)
name <- strsplit(names(MEs0),"_")
samples <-c()
for(i in 1:length(name)){
samples[[i]] <- name[[i]][2]
}  
name <- strsplit(samples,"\"")
name <- unlist(name)
colnames(MEs0) <-name
MEs <- orderMEs(MEs0)
moduleTraitCor <- cor(MEs, phenotype, use = "p");
moduleTraitPvalue <- corPvalueStudent(moduleTraitCor, nSamples)
```

## Relationship between Modules and Traits
```{r, results='asis'}
#sizeGrWindow(10,6)
# Will display correlations and their p-values
textMatrix = paste(signif(moduleTraitCor, 2), "\n(",
                        signif(moduleTraitPvalue, 1), ")", sep = "");
dim(textMatrix) = dim(moduleTraitCor)

# Display the correlation values within a heatmap plot
heat <- pheatmap(moduleTraitCor, main = paste("Module-trait relationships"), display_numbers = textMatrix,color=colorRampPalette(brewer.pal(n = 11, name = "Spectral"))(10), cluster_rows = F, cluster_cols = F, fontsize_number = 4, angle_col = 45, number_color = "black", border_color = "white")
heat
```

## Tissue

The top 10 genes that were significant for tissue within the **brown** module and had the highest correlation with brown were:

1. Gcnt4
2. Mpzl2
3. Nlrp6
4. Anks4b
5. Sh3gl2
6. Rims1
7. Atp10b
8. Sgpp2
9. Hnf4a
10. Otc

When assessing the overall sample contribution through their summary eigengenes it appears that for **brown** all intestine had a positive effect, while spleen and skeletal muscle had a negative effect. Muscle and spleen had mostly under-expressed genes while intestine had mostly over-expressed genes.

## Treatment

The top 10 genes that were significant for treatment within the **blue** module and had the highest correlation with blue were:

1. Prelid1
2. Nagk
3. Tmem8
4. Myl6
5. Tnni2
6. Sema5a
7. Fbxl19
8. Sptlc1
9. Nipsnap1
10. Cacna1s

When assessing the overall sample contribution through their summary eigengenes it appears that for **blue** all spleen and intestine had a positive effect, while skeletal muscle had a negative effect, all independent of treatment. Muscle and spleen had mostly under-expressed genes while intestine had mostly over-expressed genes.

The top 10 genes that were significant for treatment within the **brown** module and had the highest correlation with brown were:

1. Gcnt4
2. Mpzl2
3. Nlrp6
4. Anks4b
5. Sh3gl2
6. Rims1
7. Atp10b
8. Sgpp2
9. Hnf4a
10. Otc

When assessing the overall sample contribution through their summary eigengenes it appears that for **brown** all intestine had a positive effect, while spleen and skeletal muscle had a negative effect, all independent of treatment. Muscle and spleen had mostly under-expressed genes while intestine had mostly over-expressed genes.

The top 10 genes that were significant for treatment within the **turquoise** module and had the highest correlation with turquoise were:

1. Gimap1
2. Baz1b
3. Lair1
4. Lyn
5. Itgb3
6. Snx5
7. Serpina3g
8. Vav1
9. Stk10
10. Poli

The top 10 genes that were significant for treatment within the **yellow** module and had the highest correlation with yellow were:

1. Tcp11l2
2. Rcsd1
3. Cst6
4. Rbp7
5. Sh3kbp1
6. Carns1
7. Iqsec1
8. Slc35f5
9. Rev1
10. Arhgef6

When assessing the overall sample contribution through their summary eigengenes it appears that for **yellow** all intestine had a negative effect, spleen had a positive effect, all independent of treatment, and skeletal muscle had a mostly negative effect. Intestine and muscle had mostly under-expressed genes while spleen had mostly over-expressed genes.

## Treatment by tissue

The top 10 genes that were significant for treatment by tissue within the **brown** module and had the highest correlation with brown were:

1. Gcnt4
2. Mpzl2
3. Nlrp6
4. Anks4b
5. Sh3gl2
6. Rims1
7. Atp10b
8. Sgpp2
9. Hnf4a
10. Otc

When assessing the overall sample contribution through their summary eigengenes it appears that for **brown** all intestine had a positive effect, while spleen and skeletal muscle had a negative effect, all independent of treatment but not tissue. Muscle and spleen had mostly under-expressed genes while intestine had mostly over-expressed genes.

The top 10 genes that were significant for treatment by tissue within the **yellow** module and had the highest correlation with yellow were:

1. Cep170
2. Neo1
3. Dhx29
4. Thnsl2
5. Arhgef6
6. Aldh5a1
7. Rab11fip2
8. Slc9a7
9. Slc30a2
10. Cmbl

When assessing the overall sample contribution through their summary eigengenes it appears that for **yellow** all intestine had a negative effect, spleen had a positive effect, all independent of treatment, and skeletal muscle had a mostly negative effect with 2 2DG having a positive effect. Intestine and muscle had mostly under-expressed genes while spleen had mostly over-expressed genes.
