---
title: "Chang_B6_96hr_4wk 2DG GSVA plots"
author: Ann Wells
date: '`r format(Sys.Date(), "%B %d, %Y")`'
output:
 html_document:
   keep_md: TRUE
   code_folding: hide
   collapsed: no
   toc: yes
   toc_float: yes
   self_contained: no
   
---

# Introduction

```{r setup, echo=FALSE, warning=FALSE, message=FALSE}
options(stringsAsFactors = F)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(progress=FALSE)

```

This dataset contains nine tissues (heart, hippocampus, hypothalamus, kidney, liver, prefrontal cortex, skeletal muscle, small intestine, and spleen) from C57BL6/J mice that were fed 2-deoxyglucose (6g/L) through their drinking water for 96hrs or 4wks. 96hr mice were given their 2DG treatment 2 weeks after the other cohort started the 4 week treatment. The organs from the mice were harvested and processed for proteomics, metabolomics, and transcriptomics. The data in this document pertains to the transcriptomics data only. The counts that were used were FPKM normalized before being log transformed. It was determined that sample A113 had low RNAseq quality and through further analyses with PCA, MA plots, and clustering was an outlier and will be removed for the rest of the analyses performed. This document will determine which pathways are significantly altered by each module.

```{r load packages}
needed.packages <- c("tidyverse", "here", "functional", "gplots", "dplyr", "GeneOverlap", "R.utils", "reshape2","magrittr","data.table", "RColorBrewer","preprocessCore", "ARTool","emmeans", "phia", "gprofiler2","rlist", "plotly","kableExtra","foreach","doParallel","doFuture")
for(i in 1:length(needed.packages)){library(needed.packages[i], character.only = TRUE)}

source(here("source_files","WGCNA_source.R"))
```

```{r load data}
heart <- readRDS(here("Data","Heart","Chang_2DG_BL6_GSVA_pathways_heart.RData"))
small.intestine <- readRDS(here("Data","Small Intestine","Chang_2DG_BL6_GSVA_pathways_small_intestine.RData"))
spleen <- readRDS(here("Data","Spleen","Chang_2DG_BL6_GSVA_pathways_spleen.RData"))
Brain <- readRDS(here("Data","Brain","Chang_2DG_BL6_GSVA_pathways_brain.RData"))

tdata.FPKM.sample.info <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_phenotypes.RData"))

tdata.FPKM <- readRDS(here("Data","20190406_RNAseq_B6_4wk_2DG_counts_numeric.RData"))

log.tdata.FPKM <- log(tdata.FPKM + 1)
log.tdata.FPKM <- as.data.frame(log.tdata.FPKM)

log.tdata.FPKM.subset <- log.tdata.FPKM[,colMeans(log.tdata.FPKM != 0) > 0.5] 

log.tdata.FPKM.sample.info.subset <- cbind(log.tdata.FPKM.subset,tdata.FPKM.sample.info[,c(27238:27240)])

```

## Treatment
```{r, cache=TRUE}
log.tdata.FPKM.sample.info.subset.heart <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(Tissue == "Heart") %>% column_to_rownames()

log.tdata.FPKM.sample.info.subset.small.intestine <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(Tissue == "Small Intestine") %>% column_to_rownames()

Treatment <- list(cbind(as.data.frame(heart[[4]]),"Treatment" = log.tdata.FPKM.sample.info.subset.heart$Treatment), cbind(as.data.frame(small.intestine[[18]]),"Treatment" = log.tdata.FPKM.sample.info.subset.small.intestine$Treatment))

heartnames <- colnames(as.data.frame(Treatment[[1]]))
SInames <- colnames(as.data.frame(Treatment[[2]]))
name <- lst(heartnames,SInames) 

colors <- c("salmon","dodgerblue")

for (i in 1:length(Treatment)){
   for(j in 1:(dim(Treatment[[i]])[2]-1)){
      for(k in 1:length(Treatment)){
         for(m in 1:(dim(Treatment[[k]])[2]-1)){
      model<-lm(Treatment[[i]][,j] ~ Treatment[[k]][,m])
      #p <- ggplot(Treatment, aes(Treatment[[k]][,m],Treatment[[i]][,j])) + geom_point()
      #dev.new(width=3, height=2, unit="in", noRStudioGD = T)
      plot(Treatment[[k]][,m],Treatment[[i]][,j], xlab = name[[k]][m], ylab = name[[i]][j], col = colors[factor(log.tdata.FPKM.sample.info.subset.heart$Treatment)], pch = 16)
      abline(model, col = "red")
      legend(x = "topleft", legend = c("Control", "2DG"), bty = "n", pch =16, col = colors[factor(log.tdata.FPKM.sample.info.subset.heart$Treatment)])
      text(0.5,-0.5, sprintf("p-value = %3.2f", summary(model)$coefficients[2,4]))
      if(i == 1){ H = "Heart"
      mtext(side = 1, line = 2, H)}
      else {SI = "Small Intestine" 
      mtext(side = 1, line = 2, SI)}
      
      if(k == 1){ H = "Heart"
      mtext(side = 2, line = 2, H)}
      else {SI = "Small Intestine" 
      mtext(side = 2, line = 2, SI)}
}
   }
   }
}

```

<!-- ## Time -->
<!-- ```{r, cache=TRUE} -->
<!-- log.tdata.FPKM.sample.info.subset.heart <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(Tissue == "Heart") %>% column_to_rownames() -->

<!-- log.tdata.FPKM.sample.info.subset.small.intestine <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(Tissue == "Small Intestine") %>% column_to_rownames() -->

<!-- log.tdata.FPKM.sample.info.subset.spleen <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(Tissue == "Spleen") %>% column_to_rownames() -->

<!-- Treatment <- list(cbind(as.data.frame(heart[[16]]),"Treatment" = log.tdata.FPKM.sample.info.subset.heart$Treatment), cbind(as.data.frame(small.intestine[[7]]),"Treatment" = log.tdata.FPKM.sample.info.subset.small.intestine$Treatment), -->
<!-- cbind(as.data.frame(spleen[[18]]),"Treatment" = log.tdata.FPKM.sample.info.subset.spleen$Treatment), -->
<!-- cbind(as.data.frame(spleen[[29]]),"Treatment" = log.tdata.FPKM.sample.info.subset.spleen$Treatment)) -->

<!-- heartnames <- colnames(as.data.frame(Treatment[[1]])) -->
<!-- SInames <- colnames(as.data.frame(Treatment[[2]])) -->
<!-- spleennames <- colnames(as.data.frame(Treatment[[3]])) -->
<!-- spleennames2 <- colnames(as.data.frame(Treatment[[4]])) -->
<!-- name <- lst(heartnames,SInames, spleennames, spleennames2) -->

<!-- colors <- c("lightblue","darkolivegreen") -->

<!-- for (i in 1:length(Treatment)){ -->
<!--    for(j in 1:(dim(Treatment[[i]])[2]-1)){ -->
<!--       for(k in 1:length(Treatment)){ -->
<!--          for(m in 1:(dim(Treatment[[k]])[2]-1)){ -->
<!--       model<-lm(Treatment[[i]][,j] ~ Treatment[[k]][,m]) -->
<!--       plot(Treatment[[k]][,m],Treatment[[i]][,j], xlab = name[[k]][m], ylab = name[[i]][j], col = colors[factor(log.tdata.FPKM.sample.info.subset.heart$Treatment)], pch = 16) -->
<!--       abline(model, col = "red") -->
<!--       legend(x = "topleft", legend = c("Control", "2DG"), bty = "n", pch =16, col = colors[factor(log.tdata.FPKM.sample.info.subset.heart$Treatment)]) -->
<!--       text(0.5,-0.5, sprintf("p-value = %3.2f", summary(model)$coefficients[2,4])) -->

<!--       if(i == 1){ H = "Heart" -->
<!--       mtext(side = 1, line = 2, H)} -->
<!--       else if(i == 2) {SI = "Small Intestine"  -->
<!--       mtext(side = 1, line = 2, SI)} -->
<!--       else if(i == 3) {S = "Spleen"  -->
<!--       mtext(side = 1, line = 2, S)} -->
<!--       else if(i == 4) {SI = "Spleen"  -->
<!--       mtext(side = 1, line = 2, S)} -->

<!--       if(k == 1){ H = "Heart" -->
<!--       mtext(side = 1, line = 2, H)} -->
<!--       else if(k == 2) {SI = "Small Intestine"  -->
<!--       mtext(side = 1, line = 2, SI)} -->
<!--       else if(k == 3) {S = "Spleen"  -->
<!--       mtext(side = 1, line = 2, S)} -->
<!--       else if(k == 4) {SI = "Spleen"  -->
<!--       mtext(side = 1, line = 2, S)} -->
<!-- } -->
<!--    } -->
<!--    } -->
<!-- } -->
<!-- ``` -->

<!-- ## Brain -->
<!-- ```{r, cache=TRUE} -->
<!-- log.tdata.FPKM.sample.info.subset.hip.hyp.cortex <- log.tdata.FPKM.sample.info.subset %>% rownames_to_column() %>% filter(Tissue %in% c("Hippocampus","Hypothanamus","Pre-frontal Cortex")) %>% column_to_rownames() -->

<!-- Treatment <- list(cbind(as.data.frame(Brain[[12]]),"Treatment" = log.tdata.FPKM.sample.info.subset.hip.hyp.cortex$Treatment), cbind(as.data.frame(Brain[[38]]),"Treatment" = log.tdata.FPKM.sample.info.subset.hip.hyp.cortex$Treatment)) -->

<!-- brainnames <- colnames(as.data.frame(Treatment[[1]])) -->
<!-- brainnames2 <- colnames(as.data.frame(Treatment[[2]])) -->
<!-- name <- list(brainnames, brainnames2) -->

<!-- colors <- c("salmon","dodgerblue") -->

<!-- for (i in 1:length(Treatment)){ -->
<!--    for(j in 1:(dim(Treatment[[i]])[2]-1)){ -->
<!--       for(k in 1:length(Treatment)){ -->
<!--          for(m in 1:(dim(Treatment[[k]])[2]-1)){ -->
<!--       model<-lm(Treatment[[i]][,j] ~ Treatment[[k]][,m]) -->
<!--       plot(Treatment[[k]][,m],Treatment[[i]][,j], xlab = name[[k]][m], ylab = name[[i]][j], col = colors[factor(log.tdata.FPKM.sample.info.subset.hip.hyp.cortex$Treatment)], pch = 16) -->
<!--       abline(model, col = "red") -->
<!--       legend(x = "topleft", legend = c("Control", "2DG"), bty = "n", pch =16, col = colors[factor(log.tdata.FPKM.sample.info.subset.hip.hyp.cortex$Treatment)]) -->
<!--       text(0.5,-0.5, sprintf("p-value = %3.2f", summary(model)$coefficients[2,4])) -->
<!--       if(i == 1){ B = "Brain" -->
<!--       mtext(side = 1, line = 2, B)} -->
<!--       else {SI = "Brain"  -->
<!--       mtext(side = 1, line = 2, B)} -->

<!--       if(k == 1){ H = "Brain" -->
<!--       mtext(side = 2, line = 2, B)} -->
<!--       else {SI = "Brain"  -->
<!--       mtext(side = 2, line = 2, B)} -->
<!-- } -->
<!--    } -->
<!--    } -->
<!-- } -->
<!-- ``` -->